lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -312,10 +312,9 @@ public function match($token)
         $lookaheadType = $this->lexer->lookahead['type'];
 
         // short-circuit on first condition, usually types match
-        if ($lookaheadType !== $token && $token !== Lexer::T_IDENTIFIER && $lookaheadType <= Lexer::T_IDENTIFIER) {
+        if ($lookaheadType !== $token && ($token !== Lexer::T_IDENTIFIER || $lookaheadType <= Lexer::T_IDENTIFIER)) {
```

Comment:
Are there specific parser tests for this? Could need some explicit coverage for these changes, if possible (if not, simply do tell: don't need you to write a complex test for no good reason)